### PR TITLE
Fixes and Improvements to DBManager

### PR DESCRIPTION
**Pull request by vsaw**
Fixes

- https://github.com/opendatacam/opendatacam/issues/306
- https://github.com/opendatacam/opendatacam/issues/305

### CODE REVIEW

File: server/db/DBManager.js [hunk 1]
```diff
@@ -38,6 +47,14 @@ class DBManager {
     });
   }
 
+  getConnectionString() {
+    return this.connectionString;
+  }
+
+  init() {
+    this.connect(getMongoUrl());
```
Comment (vsaw) [inline server/db/DBManager.js:55]: This breaks the API I need to return the Promise generated by `connect`

File: spec/server/utils/configHelper.spec.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      };
+
+      expect(getPortFromConfig(config, 'app', 3000)).toBe(3000);
+    });
+  });
+
+  describe('process env', () => {
+    // This test fails as soon as some other test suit includes the configHelper first they will no
+    // longer use the ENV values provided here but keep the other values
+    xit('takes ENV values by default', () => {
```
Comment (vsaw) [inline spec/server/utils/configHelper.spec.js:65]: maybe I can fix the test via Jasmine helpers?

Comment (vsaw) [reply spec/server/utils/configHelper.spec.js]: Fixed in https://github.com/opendatacam/opendatacam/pull/311/commits/01d87897723829ca05b0b97cdaa42bdb6d1ade2a

**Comment by vsaw** [comment]
This turned out a bit bigger than expected but at least we bumped the number of tests from 9 to 26 ;-)

This closes #305 and #306.

File: server/db/DBManager.js [hunk 3]
```diff
@@ -1,41 +1,95 @@
...
-    });
+    } else if (isDbObject) {
+      this.db = connectionStringOrDbObject;
+      createCollectionsAndIndex(this.db);
+      return Promise.resolve(this.db);
+    } else {
+      return new Error();
+    }
+
+    this.connectionString = connectionStringOrDbObject;
```
Comment (vsaw) [inline server/db/DBManager.js:79]: This is unreachable code. Left overs that should be removed.

Comment (vsaw) [reply server/db/DBManager.js]: Line was removed in https://github.com/opendatacam/opendatacam/pull/311/commits/1da3a7ace61089d026c99c74ebf0df8fa550ca76

**Comment by tdurand** [comment]
Great ! thanks !

**Pull request merged**